src/IPHlpApi.Tests/IPHlpApiFacts.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright © .NET Foundation and Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net.NetworkInformation;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+using PInvoke;
+using Xunit;
+
+public class IPHlpApiFacts
+{
+    [Fact]
+    public void GetExtendedTcpTableTest()
+    {
+        IntPtr tcpTable = IntPtr.Zero;
+        int tcpTableLength = 0;
+
+        if (IPHlpApi.GetExtendedTcpTable(tcpTable, ref tcpTableLength, sort: true, AddressFamily.InterNetwork, IPHlpApi.TcpTableType.OwnerPidAll, 0) == Win32ErrorCode.ERROR_INSUFFICIENT_BUFFER)
+        {
+            try
+            {
+                tcpTable = Marshal.AllocHGlobal(tcpTableLength);
+
+                if (IPHlpApi.GetExtendedTcpTable(tcpTable, ref tcpTableLength, sort: true, AddressFamily.InterNetwork, IPHlpApi.TcpTableType.OwnerPidAll, 0) == Win32ErrorCode.ERROR_SUCCESS)
+                {
+                    byte[] data = new byte[tcpTableLength];
+                    Marshal.Copy(tcpTable, data, 0, tcpTableLength);
+
+                    int offset = 0;
+
+                    var table = Marshal.PtrToStructure<IPHlpApi.MIB_TCPTABLE_OWNER_PID>(tcpTable + offset);
```

Comment:
If you use pointers, you can simply cast the pointer instead of copying data as this method does.